io/module_wrt_grid_comp.F90

```diff
@@ -2106,48 +2192,66 @@ subroutine wrt_run(wrt_comp, imp_state_write, exp_state_write,clock,rc)
+                                              purpose="FV3", status=ESMF_FILESTATUS_OLD,     &
+                                              timeslice=step, state=optimize(nbdl)%state,    &
+                                              comps=optimize(nbdl)%comps, rc=rc)
+              if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+            else
+              call  write_restart_netcdf(wrt_int_state%wrtFB(nbdl),trim(filename), &
+                                  .true., wrt_mpi_comm, wrt_int_state%mype, &
+                                  rc)
+            endif ! cubed sphere vs. regional/nest write grid
+
+          else ! history bundle
           if (trim(output_grid(grid_id)) == 'cubed_sphere_grid') then
 
-            wbeg = MPI_Wtime()
             if (trim(output_file(nnnn)) == 'netcdf_parallel') then
               call write_netcdf(wrt_int_state%wrtFB(nbdl),trim(filename), &
                                .true., wrt_mpi_comm,wrt_int_state%mype, &
                                grid_id,rc)
             else
-              call ESMFproto_FieldBundleWrite(gridFB, filename=trim(filename),      &
-                                              convention="NetCDF", purpose="FV3",   &
-                                              status=ESMF_FILESTATUS_REPLACE,       &
+              call ESMFproto_FieldBundleWrite(gridFB, filename=trim(filename),               &
+                                              convention="NetCDF", purpose="FV3",            &
+                                              status=ESMF_FILESTATUS_REPLACE,                &
                                               state=stateGridFB, comps=compsGridFB,rc=rc)
 
               if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
 
-              call ESMFproto_FieldBundleWrite(wrt_int_state%wrtFB(nbdl),                    &
-                                             filename=trim(filename), convention="NetCDF",  &
-                                             purpose="FV3", status=ESMF_FILESTATUS_OLD,     &
-                                             timeslice=step, state=optimize(nbdl)%state,    &
-                                             comps=optimize(nbdl)%comps, rc=rc)
+              call ESMFproto_FieldBundleWrite(wrt_int_state%wrtFB(nbdl),                     &
+                                              filename=trim(filename), convention="NetCDF",  &
+                                              purpose="FV3", status=ESMF_FILESTATUS_OLD,     &
+                                              timeslice=step, state=optimize(nbdl)%state,    &
+                                              comps=optimize(nbdl)%comps, rc=rc)
 
               if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
 
             end if
-            wend = MPI_Wtime()
```

Comment:
I do want to keep the timing information when writing each file, it helps identify the choice of compression level etc.

Reply:
We still print the timing information. See lines 2284-2288. I just reorganized code to avoid duplication of print statements.

Reply:
I see, thanks.